lib/shopify_api/api_version.rb

```diff
@@ -1,113 +1,111 @@
 # frozen_string_literal: true
 module ShopifyAPI
-  class ApiVersion
+  class ApiVersion < ActiveResource::Base
     class UnknownVersion < StandardError; end
     class InvalidVersion < StandardError; end
-
-    extend DefinedVersions
-
     include Comparable
 
-    def self.coerce_to_version(version_or_name)
-      return version_or_name if version_or_name.is_a?(ApiVersion)
-
-      @versions ||= {}
-      @versions.fetch(version_or_name.to_s) do
-        raise UnknownVersion, "#{version_or_name} is not in the defined version set: #{@versions.keys.join(', ')}"
+    self.site = "https://app.shopify.com/services/"
+    self.primary_key = :handle
+
+    HANDLE_FORMAT = /^\d{4}-\d{2}$/.freeze
+    UNSTABLE_HANDLE = 'unstable'
+    UNSTABLE_AS_DATE = Time.utc(3000, 1, 1)
+    API_PREFIX = '/admin/api/'
+
+    class << self
+      attr_reader :versions
+
+      def coerce_to_version(version_or_handle)
+        return version_or_handle if version_or_handle.is_a?(ApiVersion)
+        if @versions.nil?
+          warn "[API VERSION WARNING] Known API Version set is empty. Initializing unvalidated version from handle."
+          return ApiVersion.new(handle: version_or_handle)
+        end
+
+        @versions[version_or_handle.to_s].tap do |api_version|
+          unless api_version
+            raise ApiVersion::UnknownVersion, "UnknownVersion API version specified, `#{version_or_handle}`. \n Versions available: #{@versions.keys}"
```

Comment:
I think this should also just be a warning, since otherwise code specifying `2019-04` will not boot after that version reaches EOL?

Reply:
well, we talked about changing the endpoint to return EOL versions as well. The current version of the gem does raise an error for unknown versions.

Reply:
I'm not sure we want the endpoint to return all EOL versions, since that would make it grow without limit.

Reply:
true, but at 4 versions a year, it'll take 12.5 years to each 50 versions, which is a pretty long time.